### DISCUSSION TITLE
git tags on snapshots

### DISCUSSION THREAD

**Discussion by evanchooly** [jreleaser/jreleaser#1463] [General]
I have a custom tag definition in my setup:  `rX.Y.Z`.  when i push a release, I get the correct tag of, say, `r1.2.3`.  If i do a release on a snapshot, i end up seeing a tag names `1.2.3`.  It seems like jreleaser is either ignoring my config on snapshot builds or I've missed a configuration somewhere but I'm not sure what that would be.  Here's what are the (hopefully) relevant parts of the config in my pom.xml:

```xml
<github>
    <releaseName>Release {{projectVersionMajor}}.{{projectVersionMinor}}.{{projectVersionPatch}}</releaseName>
    <tagName>r{{projectVersionMajor}}.{{projectVersionMinor}}.{{projectVersionPatch}}</tagName>
    <milestone>
        <name>{{projectVersionMajor}}.{{projectVersionMinor}}.{{projectVersionPatch}}</name>
    </milestone>
    <overwrite>true</overwrite>
    <update>
        <enabled>true</enabled>
        <sections>
            <section>TITLE</section>
            <section>BODY</section>
            <section>ASSETS</section>
        </sections>
    </update>
    <prerelease>
        <pattern>.*-SNAPSHOT</pattern>
    </prerelease>
</github>
```

**Comment by evanchooly** [comment]
Yes!  The rubber ducks strikes again.  I just found `project` -> `snapshot`.  I'll play with that.

**Reply by evanchooly** [reply]
piffle.  nope.  still getting `1.2.3` tagged

**Reply by evanchooly** [reply]
Ideally, there'd be no tag at all on a snapshot release but that doesn't seem configurable.

**Reply by aalmiray** [reply]
Well the default tag for a snapshot release should be `early-access` unless `project.snapshot.label` were to be overridden. What does `mvn jreleaser:config` show in the project and release sections?

**Reply by evanchooly** [reply]
I changed it because it was doing weird things on the releases page on github.  after some thrashing, i think i'm narrowing down on a workable set up.  I'm not a fan of the tagging of the snapshots but i have actions configured to ignore those tags now at least.  because I do a build for every push and then a release of that snapshot build, that extra tag was resulting in an extra CI run because the tag is also a trigger.  (I also just (re)discovered `skipTag` which is probably what I actually want.)

I'm still seeing multiple releases which is a bit of a bummer.  Honestly, I only want those built on a release build but there's no way to specify `<active>RELEASE</active>` like there is on other groups.  Adding that, i think, would eliminate 90% of my thrashing.  :)

**Reply by evanchooly** [reply]
Actually if restructure my pom a bit I can parameterize `skipRelease`...

**Reply by aalmiray** [reply]
Snapshot releases must be tagged as tags are a prerequisite for creating a GitHub release. What may happen in your CI setup is skipping a release if it's snapshot. Here's the reverse setup: early-access builds are only run if the version is snapshot https://github.com/jreleaser/helloworld-java-bin/blob/main/.github/workflows/early-access.yml

**Reply by evanchooly** [reply]
GH releases only need a git hash, actually, but that's beside the point.  I think the `skipRelease` element will get me there with a little gymnastics but having that `active` element would be far better.  And more consistent.  But I'll work with what I've got.
